# [HackBio internship 2021](https://thehackbio.com/): Genomics-One-B

![hackbio image](https://media-exp1.licdn.com/dms/image/C561BAQHKcVQGbcedOA/company-background_10000/0/1598491473588?e=2159024400&v=beta&t=rxECjvQ_YSc28Dn0n9YOtDoFFmvXjatRiqc__C2mpU0)
![](https://files.slack.com/files-pri/T025KDN24L8-F02BXHY1JMP/1629208803397.jpg)

[HackBio](https://thehackbio.com/) is a virtually regimented research internship that is practice oriented and focused on equipping African scientists with advanced bioinformatics and computational biology skills. By the end of internship, successful interns should have:
- Honed their skills in a specific bioinformatics method
- Have at least a peer-reviewed article to show for the internship experience

# PROJECT WORKFLOW & DESIGN
![hackbio ads](https://github.com/HackBio-Genomics-One-B/Genomics-One-B/blob/main/PROJECT%20DESIGN%20(GENOMICS%201B).png?raw=true)

## Step-by-Step 

### Task distribution:

|STEP|TASK|COLLABORATORS' SLACK USERNAME |PROFILE PICTURE|
|----|--------------|--------------------|---------------|
|1|Importing data (From Galaxy Library) |@Solomon @Temmykeji|******|
|2|Checking data quality using FastQC|@Solomon @Temmykeji|******|
|3|Mapping reads to a reference using BWA-MEM|@Rajeshcha44 @Nitigya-M|******|
|4|**Post-processing mapped reads** _*Merge datasets using SAMFiles_ _*Remove duplicates using MarkDuplicates_  _*Left aligning indels using BAM left align_ _*Filtering reads_|@abdnahid_ @Mike @Karteek|******|      
|5|Calling non-diploid variants using FreeBayes|@Priyacomp @MANGAIYARKARASI @pragna_lakshmi|******|
|6|Filtering variants using VCFfilte|@Naomi @Galaxy @Aarathi04|******|
|7|Visualization using IGV|@Gautami @Shreyashi @ZubairAlam|******|
|8|Compare frequencies|@omimiIII @Temmykeji @pragna_lakshmi @Gautami|******|



# Calling variants in non-diploid systems
#### _By: AvatarAnton Nekrutenko AvatarAlex Ostrovsky_

## Questions
How does frequency of mitochondrial polymorphisms change from mother to child?

## Objectives
Using Galaxy’s main site we will see how to call variants in bacteria, viruses, and organelles.

## Requirements
1. Introduction to Galaxy Analyses
2. Sequence analysis
      - Quality Control
      - Mapping

## Introduction
A handful of life ranging from prokaryotes, down to viruses and a few extension operate on non-diploid mechanism.
In this tutorial Team Genomics_One_B will be recreating the above project which involves working on four datasets, gotten from human genomic DNA sequencing. The aim of this is to identify heteroplasmies variant within the mitochondria DNA using Galaxy packages.

In this tutorial, we will cover:

STEP 1: IMPORTING DATASET
      •Download datasets from resource page
      •Click upload data on Galaxy web page
      •Galaxy will prompt to ask if it is from the local files or web (it depends on where you saved the dataset)
      •After uploading, click start. Once import is completed, the dataset highlight turns green as seen on the picture below.
      
STEP 2: QUALITY CHECK OF DATASET
It is important to check the quality of the data to be used before proceeding with the analysis. This is done to determine if there is a problem with the data setOn Galaxy, by the left hand side, click FASTA/FastqThen click FastQC Read Quality Check and execute. It will run a check on the data.

STEP 3: MAPPING THE READS USING BWA MEM 
Human genome, ‘hg38’ was used as the reference genome.Using the Paired end sequencing, the datasets has to be uploaded by selecting multiple datasets as follows:
      First set of reads: both dataset 1 (raw_child-ds-1.fq & raw_mother-ds-1.fq )
      Second set of reads: both datasets 2 (raw_child-ds-2.fq & raw_mother-ds-2.fq )
      Set read groups information to “Set read groups (SAM/BAM specification) and Execute
      
STEP 4: POST-PROCESSING MAPPED READ
      Step 4.1: Merging BAM datasetsSelect Picard from the left side of the Galaxy page and click Merge SAM Files tool, Then import dataset obtained from Step 3 into the dataset collectionInput parameters as seen in the image belowThen executeStep 3
      Step 4.2: Removing duplicates using MarkDuplicatesSelect Picard on the left side panel, click on MarkDuplicates In the SAM/BAM or dataset collection box, upload merged SAMFiles Input parameters as seen below and Execute
      Step 4.3: Left-aligning indels using BamLeftAlign toolLeft aligning of indels is important for obtaining accurate variant calls (The BAM dataset generated by MarkDuplicates will be used to run this step)Click on Variant analysis and select BamLeftAlignInput MarkDuplicate dataset and use reference genome: hg38(Input parameters as seen in picture below)Execute
      Step 4.4: Filtering reads Select filter under BamToolsUsing MarkDuplicates dataset, input parameters as seen in picture below:Then execute (NB: Would you like to set rules should be NO)
      
STEP 5: CALLING NON-DIPLOID VARIANTS USING FREEBAYESYou can navigate to the tool (FreeBayes) using the search button in GalaxySelect the reference genome mode of run and the BAM file input as followsSet the parameters for the following options using uploaded pictures as reference:Population model optionsAllelic scope optionsInput filter options

STEP 6: FILTERING VARIANTS USING VCFfilterNavigate to tool (VCFfilter) using the search button option on Galaxy webUsing the dataset obtained from step 5 Input parameters as seen in the image below and execute

STEP 7: VISUALIZATION USING IGV

STEP 8: COMPARING FREQUENCIES



1. Importing data
For this tutorial we have prepared a subset of data previously by our group (Rebolledo-Jaramillo et al. 2014). Let’s import these data into Galaxy. They are available from this Galaxy Library or via Zenodo

>
