# [HackBio internship 2021](https://thehackbio.com/): Genomics-One-B

![hackbio image](https://media-exp1.licdn.com/dms/image/C561BAQHKcVQGbcedOA/company-background_10000/0/1598491473588?e=2159024400&v=beta&t=rxECjvQ_YSc28Dn0n9YOtDoFFmvXjatRiqc__C2mpU0)
![](https://files.slack.com/files-pri/T025KDN24L8-F02BXHY1JMP/1629208803397.jpg)

[HackBio](https://thehackbio.com/) is a virtually regimented research internship that is practice oriented and focused on equipping African scientists with advanced bioinformatics and computational biology skills. By the end of internship, successful interns should have:
- Honed their skills in a specific bioinformatics method
- Have at least a peer-reviewed article to show for the internship experience

# PROJECT WORKFLOW & DESIGN
![hackbio ads](https://github.com/HackBio-Genomics-One-B/Genomics-One-B/blob/main/PROJECT%20DESIGN%20(GENOMICS%201B).png?raw=true)

## Step-by-Step 

### Task distribution:

|STEP|TASK|COLLABORATORS' SLACK USERNAME |
|----|--------------|--------------------|
|1|Importing data (From Galaxy Library) |@Solomon @Temmykeji|
|2|Checking data quality using FastQC|@Solomon @Temmykeji|
|3|Mapping reads to a reference using BWA-MEM|@Rajeshcha44 @Nitigya-M|
|4|**Post-processing mapped reads** _*Merge datasets using SAMFiles_ _*Remove duplicates using MarkDuplicates_  _*Left aligning indels using BAM left align_ _*Filtering reads_|@abdnahid_ @Mike @Karteek|     
|5|Calling non-diploid variants using FreeBayes|@Priyacomp @MANGAIYARKARASI @pragna_lakshmi|
|6|Filtering variants using VCFfilte|@Naomi @Galaxy @Aarathi04|
|7|Visualization using IGV|@Gautami @Shreyashi @ZubairAlam|
|8|Compare frequencies|@omimiIII @Temmykeji @pragna_lakshmi @Gautami|



# Calling variants in non-diploid systems
#### _By: AvatarAnton Nekrutenko AvatarAlex Ostrovsky_

## Questions
How does frequency of mitochondrial polymorphisms change from mother to child?

## Objectives
Using Galaxy‚Äôs main site we will see how to call variants in bacteria, viruses, and organelles.

## Requirements
1. Introduction to Galaxy Analyses
2. Sequence analysis
      - Quality Control
      - Mapping

## Introduction
A handful of life ranging from prokaryotes, down to viruses and a few extension operate on non-diploid mechanism.
In this tutorial Team Genomics_One_B will be recreating the above project which involves working on four datasets, gotten from human genomic DNA sequencing. The aim of this is to identify heteroplasmies variant within the mitochondria DNA using Galaxy packages.

In this tutorial, we will cover:

## STEP 1: IMPORTING DATASET

      ‚Ä¢Download datasets from resource page
      ‚Ä¢Click upload data on Galaxy web page
      ‚Ä¢Galaxy will prompt to ask if it is from the local files or web (it depends on where you saved the dataset)
      ‚Ä¢After uploading, click start. Once import is completed, the dataset highlight turns green as seen on the picture below.
      
## STEP 2: QUALITY CHECK OF DATASET

It is important to check the quality of the data to be used before proceeding with the analysis. This is done to determine if there is a problem with the data setOn Galaxy, by the left hand side, click FASTA/FastqThen click FastQC Read Quality Check and execute. It will run a check on the data.

## STEP 3: MAPPING THE READS USING BWA MEM 

Human genome, ‚Äòhg38‚Äô was used as the reference genome.Using the Paired end sequencing, the datasets has to be uploaded by selecting multiple datasets as follows:

      First set of reads: both dataset 1 (raw_child-ds-1.fq & raw_mother-ds-1.fq )
      Second set of reads: both datasets 2 (raw_child-ds-2.fq & raw_mother-ds-2.fq )
      Set read groups information to ‚ÄúSet read groups (SAM/BAM specification) and Execute
      
## STEP 4: POST-PROCESSING MAPPED READ

      Step 4.1: Merging BAM datasetsSelect Picard from the left side of the Galaxy page.
      Click Merge SAM Files tool, Then import dataset obtained from Step 3 into the dataset collection.
      Input parameters as seen in the image below. Then executeStep 3
      
      Step 4.2: Removing duplicates using MarkDuplicates
      Select Picard on the left side panel. 
      Click on MarkDuplicates In the SAM/BAM or dataset collection box. 
      Upload merged SAMFiles Input parameters as seen below and Execute
      
      Step 4.3: Left-aligning indels using BamLeftAlign 
      ToolLeft aligning of indels is important for obtaining accurate variant calls 
      (The BAM dataset generated by MarkDuplicates will be used to run this step)
      Click on Variant analysis and select BamLeftAlign
      Input MarkDuplicate dataset and 
      Use reference genome: hg38(Input parameters as seen in picture below)Execute
      
      Step 4.4: Filtering reads Select filter under BamToolsUsing MarkDuplicates dataset. 
      Input parameters as seen in picture below.
      Then execute (NB: Would you like to set rules should be NO)
      
## STEP 5: CALLING NON-DIPLOID VARIANTS USING FREEBAYES

You can navigate to the tool (FreeBayes) using the search button in GalaxySelect the reference genome mode of run and the BAM file input as followsSet the parameters for the following options using uploaded pictures as reference:ÔÇ∑Population model optionsÔÇ∑Allelic scope optionsÔÇ∑Input filter optionsÔÇ∑

## STEP 6: FILTERING VARIANTS USING VCF

FilterNavigate to tool (VCFfilter) using the search button option on Galaxy webUsing the dataset obtained from step 5 Input parameters as seen in the image below and execute

## STEP 7: VISUALIZATION USING IGV
````bash
Click on processed VCF datasets, it will expand to show link. 
Click on ‚Äúdisplay at vcf.iobio‚Äù at the bottom
Use the reference genome, Human hg38 for comparison
VCF datasets will be index to display them
Repeat process for IGV by clicking on "Display with IGV"
````
## STEP 8: COMPARING FREQUENCIES


## üìåTeam üöÄ Contribution

Individual committment to a group effort- that is what makes a team work, a company work, a society work, a civilization work - **Vince Lombardi**. Any contributions you make are **greatly appreciated** üëèüëç

**Contributors:**


  
<img src="" width="300" height="300">


- `Temmykeji` 
- Graphic design of workflow, Dataset and FastQC

      






  <p align="center">
<img src="https://user-images.githubusercontent.com/77963733/128222640-e9fb13a7-178d-42f6-8282-fcb8d27454a7.jpg" width="300" height="300">


- `Solomon` 
- Dataset, FastQC and Github Markdown


  <p align="center">
<img src="" width="300" height="300">


- `Rajeshcha44` 
- Mapping of read using BWA-MEM
      
      
  <p align="center">
<img src="https://ca.slack-edge.com/T025KDN24L8-U029R6QRT7F-5de4310f5208-512" width="300" height="300">


- `Nitigya-M` 
- Mapping of read using BWA-MEM





  <p align="center">
<img src="https://ca.slack-edge.com/T025KDN24L8-U029SQ2CZHS-ba091c5c5caa-512" width="300" height="300">


- `Abdnahid_` 
- Post Processing mapped read (This contains four steps)

      
      
  <p align="center">
<img src="" width="300" height="300">


- `Mike` 
- Post Processing mapped read (This contains four steps)



  <p align="center">
<img src="https://ca.slack-edge.com/T025KDN24L8-U02A39GJWFK-a11377b2a1f0-512" width="300" height="300">


- `Karteek` 
- Post Processing mapped read (This contains four steps)
      


      
  <p align="center">
<img src="https://ca.slack-edge.com/T025KDN24L8-U02AGS5QS00-6a6af014db65-512" width="300" height="300">


- `Priyacomp` 
- Variant calling of dataset
      
      
  <p align="center">
<img src="https://ca.slack-edge.com/T025KDN24L8-U029NG6R8UV-bb016f49b6c9-512" width="300" height="300">


- `MANGAIYARKARASI` 
- Variant calling of dataset

      
  <p align="center">
<img src="https://ca.slack-edge.com/T025KDN24L8-U029C9A9J79-f95d8e9a01dc-512" width="300" height="300">


- `Pragna_lakshmi` 
- Variant calling of dataset using FreeBayes and Comparing of frequencies using VCFtoTab-delimited
        
        
  <p align="center">
<img src="https://ca.slack-edge.com/T025KDN24L8-U029S3A3TMH-2f863dfd6528-512" width="300" height="300">


- `Naomi` 
- Mapping of read using BWA-MEM


   <p align="center">
<img src="" width="300" height="300">


- `Galaxy` 
- Filtering of variant call dataset using FreeBayes


  <p align="center">
<img src="" width="300" height="300">


- `Aarathi04` 
- Filtering of variant call dataset using VCFfilter

       
        
        
        

    
  <p align="center">
<img src="https://ca.slack-edge.com/T025KDN24L8-U029WT4Q31S-723b12b5b8d5-512" width="300" height="300">
      
           
- `Gautami`**(Team Leader)**
- Visualization using IGV and VCF.IOBIO; and Comparing of frequencies using VCFtoTab-delimited



  <p align="center">
<img src="https://ca.slack-edge.com/T025KDN24L8-U02AEQNK4JC-d01d2eb3a8bf-512" width="300" height="300">


- `ZubairAlam` 
- Visualization using IGV and VCF.IOBIO


  <p align="center">
<img src="" width="300" height="300">


- `Shreyashi` 
- Visualization using IGV and VCF.IOBIO


  <p align="center">
<img src="https://ca.slack-edge.com/T025KDN24L8-U029ME14L3G-b45b38f03dfc-512" width="300" height="300">


- `omimill` 
- Comparing of frequencies using VCFtoTab-delimited and Github Markdown
        
        
        
        
<p align="center">
<img src="https://user-images.githubusercontent.com/77963733/129745575-ecdb0b4d-981f-435b-bd56-bc3ed9da4a44.png" width="500" heigth="500">


